bfsbdump

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python3
+
+"""
+Dump BlueField secure boot status information.
+"""
+
+import argparse
+import struct
+import binascii
+import os
+import sys
+import subprocess
+
+LifeCycleState = [ 'Production', 'Secure', 'Non-Secure', 'RMA' ]
+
+def GetPlatformName():
+    Out = subprocess.run(['bfhcafw', 'mcra', '0xf0014.0:16'], stdout=subprocess.PIPE)
+    DeviceId = Out.stdout[:-1].decode('utf-8')
+    if DeviceId == '0x00000211':
+        return 'BlueField1'
+    elif DeviceId == '0x00000214':
+        return 'BlueField2'
+    elif DeviceId == '0x0000021c':
+        return 'BlueField3'
+    else:
+        printf('Error! can not read device id')
+        sys.exit (1)
+
+def CountSetBits(n):
+    if n == 0:
+        return 0
+    else:
+        return (n & 1) + CountSetBits(n >> 1)
+
+class ChipSecureBootStatus:
+
+    def __init__(self):
+        self.PlatformName = GetPlatformName()
+
+    def Decode(self, data):
+        __StructFormat = '<I32sIHHBBBBB??B?15s'
+        __StructSize = struct.calcsize(__StructFormat)
+        if __StructSize != len(data):
+            print('Error! bad size {}, expected size {}'.format(__StructSize, len(data)))
+            sys.exit (1)
+
+        '''
+        Defined C structure which encapsulates the BlueField Secure Boot
+        status information.
+
+        typedef struct {
+        UINT64  DotChipId[4];
+        UINT32  DotFsblVersion;
+        UINT16  DotMonotonicCounter;
+        UINT16  RevMonotonicCounter;
+        UINT8   TrustedBootFwNonVolatileCounter;
+        UINT8   UntrustedBootFwNonVolatileCounter;
+        UINT8   CerberusNonVolatileCounter;
+        UINT8   NvProductionStatus;
+        UINT8   LifeCycleStatus;
+        BOOLEAN SecureBootEnabled;
+        BOOLEAN SecureBootEnabledWithDevKey;
+        UINT8   SecureBootKeyStatus;
+        BOOLEAN CryptoEnabled;
+        UINT8   Reserved[15];
+        } CHIP_SECURE_BOOT_STATUS;
+        '''
+        (self.__hdr,
+            self.DotChipId,
+            self.DotFsblVersion,
+            self.DotMonotonicCounter,
+            self.RevMonotonicCounter,
+            self.TrustedBootFwNonVolatileCounter,
+            self.UntrustedBootFwNonVolatileCounter,
+            self.CerberusNonVolatileCounter,
+            self.NvProduction,
+            self.LifeCycle,
+            self.SecureBootEnabled,
+            self.SecureBootEnabledWithDevKey,
+            self.SecureBootKeyStatus,
+            self.CryptoEnabled,
+            self.Reserved) = struct.unpack(__StructFormat, data)
+    
+    def Dump(self):
+        if self.LifeCycle > 3:
+            print('Error! bad life cycle state {:02X}, expected {:02X}, {:02X}, {:02X}, or {:02X}'.format(
+                    self.LifeCycle, 0, 1, 2, 3 ))
```

Comment:
Same as above, should be printed to `stdout`

Reply:
done